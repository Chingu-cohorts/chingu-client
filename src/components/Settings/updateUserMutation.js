import gql from "graphql-tag";

const updateUser = gql`
  mutation updateUser(
    $first_name: String!
    $last_name: String!
    $bio: String
    $github_url: String!
    $twitter_url: String
    $linkedin_url: String
    $portfolio_url: String
    $website_url: String
    $blog_url: String
  ) {
    updateUser(
      user_data: {
        first_name: $first_name
        last_name: $last_name
        bio: $bio
        github_url: $github_url
        twitter_url: $twitter_url
        linkedin_url: $linkedin_url
        portfolio_url: $portfolio_url
        website_url: $website_url
        blog_url: $blog_url
      }
    ) {
      first_name
    }
  }
`;

export default updateUser;
