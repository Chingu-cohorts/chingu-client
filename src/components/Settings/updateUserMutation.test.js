import updateUserMutation from "./updateUserMutation";

describe("updateUserMutation", () => {
  it("should be the correct mutation", () => {
    expect(updateUserMutation).toMatchSnapshot();
  });
});
