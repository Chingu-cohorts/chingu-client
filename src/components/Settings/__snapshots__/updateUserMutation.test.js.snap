// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`updateUserMutation should be the correct mutation 1`] = `
Object {
  "definitions": Array [
    Object {
      "directives": Array [],
      "kind": "OperationDefinition",
      "name": Object {
        "kind": "Name",
        "value": "updateUser",
      },
      "operation": "mutation",
      "selectionSet": Object {
        "kind": "SelectionSet",
        "selections": Array [
          Object {
            "alias": null,
            "arguments": Array [
              Object {
                "kind": "Argument",
                "name": Object {
                  "kind": "Name",
                  "value": "user_data",
                },
                "value": Object {
                  "fields": Array [
                    Object {
                      "kind": "ObjectField",
                      "name": Object {
                        "kind": "Name",
                        "value": "first_name",
                      },
                      "value": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "first_name",
                        },
                      },
                    },
                    Object {
                      "kind": "ObjectField",
                      "name": Object {
                        "kind": "Name",
                        "value": "last_name",
                      },
                      "value": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "last_name",
                        },
                      },
                    },
                    Object {
                      "kind": "ObjectField",
                      "name": Object {
                        "kind": "Name",
                        "value": "bio",
                      },
                      "value": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "bio",
                        },
                      },
                    },
                    Object {
                      "kind": "ObjectField",
                      "name": Object {
                        "kind": "Name",
                        "value": "github_url",
                      },
                      "value": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "github_url",
                        },
                      },
                    },
                    Object {
                      "kind": "ObjectField",
                      "name": Object {
                        "kind": "Name",
                        "value": "twitter_url",
                      },
                      "value": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "twitter_url",
                        },
                      },
                    },
                    Object {
                      "kind": "ObjectField",
                      "name": Object {
                        "kind": "Name",
                        "value": "linkedin_url",
                      },
                      "value": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "linkedin_url",
                        },
                      },
                    },
                    Object {
                      "kind": "ObjectField",
                      "name": Object {
                        "kind": "Name",
                        "value": "portfolio_url",
                      },
                      "value": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "portfolio_url",
                        },
                      },
                    },
                    Object {
                      "kind": "ObjectField",
                      "name": Object {
                        "kind": "Name",
                        "value": "website_url",
                      },
                      "value": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "website_url",
                        },
                      },
                    },
                    Object {
                      "kind": "ObjectField",
                      "name": Object {
                        "kind": "Name",
                        "value": "blog_url",
                      },
                      "value": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "blog_url",
                        },
                      },
                    },
                  ],
                  "kind": "ObjectValue",
                },
              },
            ],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "updateUser",
            },
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "first_name",
                  },
                  "selectionSet": null,
                },
              ],
            },
          },
        ],
      },
      "variableDefinitions": Array [
        Object {
          "defaultValue": null,
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NonNullType",
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "String",
              },
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "first_name",
            },
          },
        },
        Object {
          "defaultValue": null,
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NonNullType",
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "String",
              },
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "last_name",
            },
          },
        },
        Object {
          "defaultValue": null,
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NamedType",
            "name": Object {
              "kind": "Name",
              "value": "String",
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "bio",
            },
          },
        },
        Object {
          "defaultValue": null,
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NonNullType",
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "String",
              },
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "github_url",
            },
          },
        },
        Object {
          "defaultValue": null,
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NamedType",
            "name": Object {
              "kind": "Name",
              "value": "String",
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "twitter_url",
            },
          },
        },
        Object {
          "defaultValue": null,
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NamedType",
            "name": Object {
              "kind": "Name",
              "value": "String",
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "linkedin_url",
            },
          },
        },
        Object {
          "defaultValue": null,
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NamedType",
            "name": Object {
              "kind": "Name",
              "value": "String",
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "portfolio_url",
            },
          },
        },
        Object {
          "defaultValue": null,
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NamedType",
            "name": Object {
              "kind": "Name",
              "value": "String",
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "website_url",
            },
          },
        },
        Object {
          "defaultValue": null,
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NamedType",
            "name": Object {
              "kind": "Name",
              "value": "String",
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "blog_url",
            },
          },
        },
      ],
    },
  ],
  "kind": "Document",
  "loc": Object {
    "end": 606,
    "start": 0,
  },
}
`;
